src/types.ts

```diff
@@ -7,6 +7,13 @@ export interface QueryInfo {
     queryName?: string;
 }
 
+export interface MultipleQueriesInfo {
+    loadedQueryName: string;
```

Comment:
Instead of having this props you can use the QueryInfo type, you just need to omit the query mashup.

loadedQueryInfo:  Omit<QueryInfo, 'queryMashup' >

Reply:
maybe instead of the list of names and formulas, better to have as input an array of queryInfo without omit refresh on open.

Reply:
We also need queryName not to be optional, is that too many changes to QueryInfo?